src/seq/nucleotide.jl

```diff
@@ -200,13 +212,30 @@ NucleotideSequence{T<:Nucleotide}(::Type{T}) = NucleotideSequence{T}(zeros(Uint6
 
 Construct a subsequence of the given type from another nucleotide sequence
 """
-function NucleotideSequence{T<:Nucleotide}(::Type{T}, other::NucleotideSequence, part::UnitRange)
+# Construct a subsequence of another nucleotide sequence
+function NucleotideSequence{T<:Nucleotide}(::Type{T}, other::NucleotideSequence,
+                                           part::UnitRange; mutable::Bool=false)
     start = other.part.start + part.start - 1
     stop = start + length(part) - 1
     if start < other.part.start || stop > other.part.stop
         error("Invalid subsequence range")
     end
-    return NucleotideSequence{T}(other.data, other.ns, start:stop)
+
+    seq = NucleotideSequence{T}(other.data, other.ns, start:stop, mutable, true)
+
+    if other.mutable || mutable
+        orphan!(seq, true)
+    end
+
+    if !other.mutable
+        other.hasrelatives = true
```

Comment:
We could simplify this in the same way you did in the Aminoacid Sequence constructor:

```julia
other.hasrelatives = !other.mutable
seq.hasrelatives = !mutable
```